### ISSUE TITLE
VIVO-2008: Generate a unique VIVO identifier from a SPARQL query

### ISSUE THREAD

**Issue by chenejac** [vivo-project/VIVO#3594]
[Michel Héon](https://github.com/MichelHeon) (Migrated from [VIVO-2008](https://vivo-project.atlassian.net/browse/VIVO-2008)) said:

The purpose of this feature is to provide the ability to obtain a unique VIVO identifier from a SPARQL query. This feature will be useful for 'query' and 'update' type requests.

Here are some examples of use:

*Verify the existence of an IRI*
```bash 
PREFIX sfnc: <http://vivoweb.org/sparql/function#>
 SELECT * 
 WHERE {
   BIND (IRI("http://localhost:8080/vivo/individual/n1046") as ?IRI) .
   ?IRI sfnc:isIriExist ?isExist .
} 
``` 

*Create Uniq IRI and assing it to a FacultyMember wtih construct query*
```bash 
PREFIX sfnc: <http://vivoweb.org/sparql/function#>
 CONSTRUCT {
    ?newIRI a vivo:FacultyMember .
 } 
 WHERE {
   "http://localhost:8080/vivo/individual/n" sfnc:hasNewIRI ?newIRI .
}
```

**Comment by litvinovg** [comment]
I think we should avoid creating SPARQL extensions as people will most probably wont use it and stick with using standard struuid() instead of creating custom function.
Currently used creating uri schema <n+number> doesn't provide randomized uri which can be a problem in some cases.
I think it would be better to solve the problem of creating a unique identifier and a meaningfull name in the URL separately.

**Comment by michel-heon** [comment]
Hi @litvinovg ,
The use case of this function is part of the data ingestion process in VIVO by a robot. 
It is about being able to create a randomized IRI the way that VIVO creates IRIs in edit mode whose structure of the Vivo IRI is precisely of the form` <http://base/n+random>`. 

The algorithm used to generate the `random` number is the same one used by VIVO
In this context, the use of `struuid()` although it generates a random number, is absolutely not suitable because the structure of the number generated is completely different from the one generated by VIVO which in some cases is a conformance obstacle,

In short, I think it is better to offer a choice to the community between two possibilities (`struuid() `and `hasNewIRI`) and that there are no facts to support the claim that the function will not be used by the community, especially since the use of `hasNewIRI `lightens the query syntax compared to the use of `struuid `(see the message of Oct 19). 
Try it to convince yourself. You'll see ;-).
